### PR TITLE
✨ Allow ingesting existing data in the cloud

### PR DESCRIPTION
**Pull request by sunnyosun**
Together with: https://github.com/laminlabs/lnschema-core/pull/96

### CODE REVIEW

File: lamindb/_record.py [hunk 1]
```diff
@@ -32,24 +34,33 @@
...
-    data: Union[Path, str, pd.DataFrame, ad.AnnData], name, format
-) -> Tuple[Any, Path, str, str]:
+    data: Union[Path, CloudPath, str, pd.DataFrame, ad.AnnData], name, format
+) -> Tuple[Any, Union[Path, CloudPath], str, str]:
     """Serialize a data object that's provided as file or in memory."""
     memory_rep = None
-    if isinstance(data, (Path, str)):
-        local_filepath = Path(data)
-        name, suffix = get_name_suffix_from_filepath(local_filepath)
+    if isinstance(data, str):
```
Comment (falexwolf) [inline lamindb/_record.py:41]: Can you add a comment: # convert str to either Path or CloudPath?

Comment (sunnyosun) [reply lamindb/_record.py:41]: ok!

File: lamindb/_record.py [hunk 2]
```diff
@@ -32,24 +34,33 @@
...
+        try:
+            data = CloudPath(data)
```
Comment (falexwolf) [inline lamindb/_record.py:43]: I think this should error here if the returned cloudpath object doesn't equate the one in settings

Hence, e.g., the following code block
```
from lndb_setup import settings
if data != settings.instance.storage.root:
    raise ValueError(
        "Can only track objects in configured cloud storage locations. Please call `lndb_setup.set_storage("<bucket_name>")`.")
```

Comment (sunnyosun) [reply lamindb/_record.py]: Done!

**Pull request merged**